src/swap.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Problem } from "./cnd/axios_rfc7807_middleware";
+import { Cnd } from "./cnd/cnd";
+import { Swap, WalletError } from "./swap";
+
+describe("Swap", () => {
+  it("Throws a problem if cnd returns an error when executing an action", async () => {
+    const basePath = "http://example.com";
+    const cnd = new Cnd(basePath);
+
+    const selfPath = "/swap-id";
+
+    const swap = new Swap(cnd, selfPath, {});
+
+    cnd.fetch = jest.fn(async () => {
+      throw new Problem({ title: "" });
+    });
+
+    const promise = swap.tryExecuteSirenAction("action", {
+      maxTimeoutSecs: 1,
+      tryIntervalSecs: 0.01
+    });
+
+    await expect(promise).rejects.toBeInstanceOf(Problem);
+  });
+
+  it("Throws a WalletError if a wallet returns an error when resolving fields", async () => {
+    const basePath = "http://example.com";
+    const cnd = new Cnd(basePath);
+
+    const swap = new Swap(cnd, `${basePath}/swap-id`, {});
+
+    // @ts-ignore: This should return an AxiosPromise
+    cnd.fetch = jest.fn(async () => {
+      return {
+        data: {
+          actions: [
+            {
+              href: "/swap-id/action",
+              name: "action",
+              method: "GET",
+              fields: [{ name: "address", class: ["bitcoin", "address"] }]
+            }
+          ]
+        }
+      };
+    });
+
+    const promise = swap.tryExecuteSirenAction("action", {
+      maxTimeoutSecs: 1,
+      tryIntervalSecs: 0.01
+    });
+
+    await expect(promise).rejects.toBeInstanceOf(WalletError);
+  });
+
+  it("Throws a WalletError if a wallet returns an error when executing an action", async () => {
```

Comment:
Which code path we trying to test here? This one? https://github.com/comit-network/comit-js-sdk/pull/152/files#diff-7ca577257aac4fa5e3c549c84277afc6R210-R218

Just make the EthereumWallet throw and exception, why tinker around with `Cnd`?

Reply:
Unfortunately your link does not work for me. Are you talking about testing `doLedgerAction` directly?

Reply:
I am talking about making the actual wallet action throw an exception and assert that is correctly wrapped in a `WalletError`.

Reply:
It does throw an exception as it is not initialised but I understand your point as it is not explicit in the test.